moto/textract/models.py

```diff
@@ -16,12 +19,50 @@ class TextractJobStatus:
 
 
 class TextractJob(BaseModel):
-    def __init__(self, job: Dict[str, Any]):
+    def __init__(
+        self, job: Dict[str, Any], notification_channel: Optional[Dict[str, str]] = None
+    ):
         self.job = job
+        self.notification_channel = notification_channel
+        self.job_id = str(mock_random.uuid4())
 
     def to_dict(self) -> Dict[str, Any]:
         return self.job
 
+    def send_completion_notification(
+        self, account_id: str, region_name: str, document_location: Dict[str, Any]
+    ) -> None:
+        if not self.notification_channel:
+            return
+
+        topic_arn = self.notification_channel.get("SNSTopicArn")
+        if not topic_arn:
+            return
+
+        # Convert document_location from {'S3Object': {'Bucket': '...', 'Name': '...'}} format
+        # to {'S3Bucket': '...', 'S3ObjectName': '...'} format as per AWS docs
+        s3_object = document_location.get("S3Object", {})
+        doc_location = {
+            "S3Bucket": s3_object.get("Bucket", ""),
+            "S3ObjectName": s3_object.get("Name", ""),
+        }
+
+        notification = {
+            "JobId": self.job_id,
+            "Status": self.job["JobStatus"],
+            "API": "StartDocumentTextDetection",
+            "JobTag": "",  # Not implemented yet
+            "Timestamp": int(time.time() * 1000),  # Convert to milliseconds
+            "DocumentLocation": doc_location,
+        }
+
+        sns_backend = sns_backends[account_id][region_name]
+        sns_backend.publish(
```

Comment:
Can we catch any `moto.sns.exceptions.TopicNotFound`-exceptions here? At this point we don't know whether the topic exists, and if it doesn't, we should IMO ignore it for now.

In the future we can add a check whether the Topic exists before starting the Job, but as long as we don't have that, I think we should just ignore invalid input.